test/cluster/test_view_build_status.py

```diff
@@ -541,3 +541,30 @@ async def test_migration_on_existing_raft_topology(request, manager: ManagerClie
         log = await manager.server_open_log(srv.server_id)
         res = await log.grep(r'ERROR.*\[shard [0-9]: [a-z]+\] raft_topology - topology change coordinator fiber got error exceptions::unavailable_exception \(Cannot achieve consistency level for')
         assert len(res) == 0
+
+# Regression test for scylladb/scylladb#23536
+# New node addition with 'sleep_in_synchronize' injection shoehorns the execution path to
+# hit 'raft_group0_client::start_operation' when upgrade state is synchronize. The test
+# confirms successful execution in such cases.
+@skip_mode('release', 'error injections are not supported in release mode')
+@pytest.mark.asyncio
+async def test_view_build_status_with_synchronize_wait(manager: ManagerClient):
+    node_count = 4
+    servers = await manager.servers_add(node_count)
+    cql, hosts = await manager.get_ready_cql(servers)
+
+    ks = await create_keyspace(cql)
+    await create_table(cql, ks)
+    await create_mv(cql, ks, "vt1")
```

Comment:
It's completely non-obvious that `start_operation` you describe in the comment above comes from creating a materialized view. Add a comment here or extend the comment above the test.